tests/test_offchain_types.py

```diff
@@ -307,6 +311,360 @@ def test_validate_write_once_fields_raises_overwrite_error_when_set_value_to_non
+
+    request_json = """{
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command_type": "PaymentCommand",
+  "command": {
+    "_ObjectType": "PaymentCommand"
+  }
+}"""
+    with pytest.raises(offchain.FieldError, match="missing field: _ObjectType") as e:
+        offchain.from_json(request_json)
+
+    assert e.value.code == "missing-field"
+
+
+def test_invalid_enum_field_value():
+    request_json = """{
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command_type": "PaymentCommand",
+  "command": {
+    "_ObjectType": "PaymentCommand",
+    "payment": {
+      "reference_id": "2185027f-0574-6f55-2668-3a38fdb5de98",
+      "sender": {
+        "address": "lbr1p7ujcndcl7nudzwt8fglhx6wxnvqqqqqqqqqqqqelu3xv",
+        "status": {
+          "status": "invalid-status-value"
+        },
+        "kyc_data": {
+          "type": "individual",
+          "payload_version": 1,
+          "given_name": "Jack",
+          "surname": "G",
+          "address": {
+            "city": "San Francisco"
+          }
+        }
+      },
+      "receiver": {
+        "address": "lbr1p7ujcndcl7nudzwt8fglhx6wxnvqqqqqqqqqqqqelu3xv",
+        "status": {
+          "status": "none"
+        }
+      },
+      "action": {
+        "amount": 1000000000000,
+        "currency": "XUS",
+        "action": "charge",
+        "timestamp": 1609391272
+      }
+    }
+  },
+  "_ObjectType": "CommandRequestObject"
+}"""
+    with pytest.raises(
+        offchain.FieldError,
+        match="expect one of \\['none', 'needs_kyc_data', 'ready_for_settlement', 'abort', 'soft_match'\\], but got: invalid-status-value",
+    ) as e:
+        offchain.from_json(request_json)
+
+    assert e.value.code == "invalid-field-value"
+
+
+def test_invalid_uuid_field_value():
+    request_json = """{
+  "cid": "invalid",
+  "command_type": "PaymentCommand",
+  "command": {
+    "_ObjectType": "PaymentCommand",
+    "payment": {
+      "reference_id": "2185027f-0574-6f55-2668-3a38fdb5de98",
+      "sender": {
+        "address": "lbr1p7ujcndcl7nudzwt8fglhx6wxnvqqqqqqqqqqqqelu3xv",
+        "status": {
+          "status": "needs_kyc_data"
+        },
+        "kyc_data": {
+          "type": "individual",
+          "payload_version": 1,
+          "given_name": "Jack",
+          "surname": "G",
+          "address": {
+            "city": "San Francisco"
+          }
+        }
+      },
+      "receiver": {
+        "address": "lbr1p7ujcndcl7nudzwt8fglhx6wxnvqqqqqqqqqqqqelu3xv",
+        "status": {
+          "status": "none"
+        }
+      },
+      "action": {
+        "amount": 1000000000000,
+        "currency": "XUS",
+        "action": "charge",
+        "timestamp": 1609391272
+      }
+    }
+  },
+  "_ObjectType": "CommandRequestObject"
+}"""
+    with pytest.raises(
+        offchain.FieldError,
+        match="invalid does not match pattern",
+    ) as e:
+        offchain.from_json(request_json)
+
+    assert e.value.code == "invalid-field-value"
```

Comment:
Is there a way to see which field value is invalid?

Reply:
added field name (with path) assertion